control/plugin/client/grpc.go

```diff
@@ -0,0 +1,213 @@
+/*
+http://www.apache.org/licenses/LICENSE-2.0.txt
+
+
+Copyright 2015 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"crypto/rsa"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"google.golang.org/grpc"
+
+	"golang.org/x/net/context"
+
+	"github.com/intelsdi-x/snap/control/plugin"
+	"github.com/intelsdi-x/snap/control/plugin/cpolicy"
+	"github.com/intelsdi-x/snap/control/plugin/encrypter"
+	"github.com/intelsdi-x/snap/control/plugin/rpc"
+	"github.com/intelsdi-x/snap/core"
+	"github.com/intelsdi-x/snap/core/ctypes"
+	"github.com/intelsdi-x/snap/grpc/common"
+	"github.com/intelsdi-x/snap/pkg/rpcutil"
+)
+
+type grpcClient struct {
+	collector  rpc.CollectorClient
+	plugin     rpc.PluginClient
+	pluginType plugin.PluginType
+	timeout    time.Duration
+	conn       *grpc.ClientConn
+	encrypter  *encrypter.Encrypter
+}
+
+// NewCollectorGrpcClient returns a collector gRPC Client.
+func NewCollectorGrpcClient(address string, timeout time.Duration, pub *rsa.PublicKey, secure bool) (PluginCollectorClient, error) {
+	stuff := strings.Split(address, ":")
+	if len(stuff) != 2 {
+		return nil, fmt.Errorf("bad address")
+	}
+	address = stuff[0]
+	port, err := strconv.ParseInt(stuff[1], 10, 64)
+	if err != nil {
+		return nil, err
+	}
+	p, err := newGrpcClient(address, int(port), timeout, plugin.CollectorPluginType)
+	if err != nil {
+		return nil, err
+	}
+	if secure {
+		key, err := encrypter.GenerateKey()
+		if err != nil {
+			return nil, err
+		}
+		encrypter := encrypter.New(pub, nil)
+		encrypter.Key = key
+		p.encrypter = encrypter
+	}
+
+	return p, nil
+}
+
+// func newGrpcClient(addr string, port int, pub *rsa.PublicKey, secure bool) (*grpcClient, error) {
+func newGrpcClient(addr string, port int, timeout time.Duration, typ plugin.PluginType) (*grpcClient, error) {
+	conn, err := rpcutil.GetClientConnection(addr, port)
+	if err != nil {
+		return nil, err
+	}
+	p := &grpcClient{
+		plugin:  rpc.NewPluginClient(conn),
+		timeout: timeout,
+		conn:    conn,
+	}
+
+	switch typ {
+	case plugin.CollectorPluginType:
+		p.collector = rpc.NewCollectorClient(conn)
+	}
```

Comment:
Do we want, for now, to have a default that will fail to load a processor or publisher, and state that it is not yet supported?  Maybe with a TODO to address it with the content-type refactor?